homeassistant/helpers/config_entry_flow.py

```diff
@@ -6,7 +6,7 @@
 from typing import TYPE_CHECKING, Any, Generic, TypeVar, Union, cast
 
 from homeassistant import config_entries
-from homeassistant.components import dhcp, mqtt, onboarding, ssdp, zeroconf
+from homeassistant.components import onboarding
```

Comment:
Since we are late in the beta cycle, I suggest we do the smallest targeted change to resolve the issue in https://github.com/home-assistant/core/pull/74464 

I'd still like to move all the service_info types into their own modules so we this problem isn't easy to accidentally recreate as we already have type checking guard in another place for this as well.